src/coreclr/src/vm/class.h

```diff
@@ -2149,14 +1956,13 @@ class LayoutEEClass : public EEClass
 {
 public:
     EEClassLayoutInfo m_LayoutInfo;
+    PTR_EEClassNativeLayoutInfo m_nativeLayoutInfo;
```

Comment:
This may better be `Volatile<PTR_EEClassNativeLayoutInfo>`. 

Without volatile, the C++ compiler can do surprise you by, e.g. changing this:

```
if (m_nativeLayoutInfo == NULL)
{
   .... initialize m_nativeLayoutInfo ...
}
return m_nativeLayoutInfo;
```

into this:

```
auto temp = m_nativeLayoutInfo;
if (m_nativeLayoutInfo != NULL)
     return temp;
.... initialize m_nativeLayoutInfo ...
return m_nativeLayoutInfo;
```

We had many bugs like that over the years...